include/dd/Node.hpp

```diff
@@ -33,39 +32,79 @@ namespace dd {
+        // 8 = marks a reduced dm node,
+        // 4 = marks a dm (tmp flag),
+        // 2 = mark first path edge (tmp flag),
+        // 1 = mark path is conjugated (tmp flag))
 
         static mNode            terminalNode;
         constexpr static mNode* terminal{&terminalNode};
 
         static constexpr bool isTerminal(const mNode* p) { return p == terminal; }
+
+        [[nodiscard]] inline bool isIdentity() const { return flags & 16; }
+        [[nodiscard]] inline bool isSymmetric() const { return flags & 32; }
+        inline void               setIdentity(bool identity) {
+            if (identity)
+                flags = (flags | 16);
+            else
+                flags = (flags & (~16));
+        }
+        inline void setSymmetric(bool symmetric) {
+            if (symmetric)
+                flags = (flags | 32);
+            else
+                flags = (flags & (~32));
+        }
     };
     using mEdge       = Edge<mNode>;
     using mCachedEdge = CachedEdge<mNode>;
 
-    inline mNode mNode::terminalNode{{{{nullptr, Complex::zero}, {nullptr, Complex::zero}, {nullptr, Complex::zero}, {nullptr, Complex::zero}}}, nullptr, 0U, -1, 0, true, true};
+    inline mNode mNode::terminalNode{{{{nullptr, Complex::zero}, {nullptr, Complex::zero}, {nullptr, Complex::zero}, {nullptr, Complex::zero}}}, nullptr, 0U, -1, 32 + 16};
 
     struct dNode {
         std::array<Edge<dNode>, NEDGE> e{};    // edges out of this node
         dNode*                         next{}; // used to link nodes in unique table
         RefCount                       ref{};  // reference count
         Qubit                          v{};    // variable index (nonterminal) value (-1 for terminal)
-        std::int_fast8_t               flags = 0;
+        std::uint_least8_t             flags = 0;
+        // 32 = marks a node with is symmetric.
+        // 16 = marks a node resembling identity
         // 8 = marks a reduced dm node,
         // 4 = marks a dm (tmp flag),
         // 2 = mark first path edge (tmp flag),
         // 1 = mark path is conjugated (tmp flag))
 
         static dNode            terminalNode;
         constexpr static dNode* terminal{&terminalNode};
-
-        static constexpr bool isTerminal(const dNode* p) { return p == terminal; }
-        //static constexpr bool isDensity() { return true; }
+        static constexpr bool   isTerminal(const dNode* p) { return p == terminal; }
     };
     using dEdge       = Edge<dNode>;
     using dCachedEdge = CachedEdge<dNode>;
     inline dNode dNode::terminalNode{{{{nullptr, Complex::zero}, {nullptr, Complex::zero}, {nullptr, Complex::zero}, {nullptr, Complex::zero}}}, nullptr, 0, -1, 0};
 
+    template<>
+    constexpr bool dEdge::operator==(const dEdge& other) const {
+        assert(p != nullptr && other.p != nullptr);
+        return p == other.p && tempDensityMatrixFlagsEqual(p->flags, other.p->flags) && w.approximatelyEquals(other.w);
+    }
+
```

Comment:
It would make much more sense to define `operator==` for the node classes and have a generic `operator==` implementation for the Edge class that just compares the nodes and the weight.

Reply:
Removed the function.